src/go/rpk/pkg/cli/cluster/partitions/list.go

```diff
@@ -80,26 +89,39 @@ List all in json format.
 			var mu sync.Mutex
 			if len(topics) == 0 && all {
 				clusterPartitions, err = cl.AllClusterPartitions(cmd.Context(), true, disabledOnly)
+				// If the admin API returns a 404, most likely rpk is talking
+				// with an old cluster.
+				if he := (*adminapi.HTTPResponseError)(nil); errors.As(err, &he) {
+					if he.Response.StatusCode == http.StatusNotFound {
+						out.Die("unable to query all partitions in the cluster: %v You may need to upgrade the cluster to access this feature or try listing per-topic with: 'rpk cluster partition list [TOPICS]'", err)
```

Comment:
tiny format nit, change `%v You` to `%v; you`, and maybe add `...` after `TOPICS` (so it looks like `[TOPICS...]` although I guess that would mean a corresponding ... on line R38?

Reply:
w.r.t adding the `;` I had this first, however, the admin API error contains a new line at the end so I ended up having:

```
unable to query all partitions in the cluster: request GET http://127.0.0.1:9644/v1/cluster/partitions?with_internal=true&disabled=false failed: Not Found, body: "{\"message\": \"Not found\", \"code\": 404}"
; You may need to upgrade the cluster to access this feature or try listing per-topic with: 'rpk cluster partition list [TOPICS]'

```

That's why I omitted the semicolon. Actually, I think that I need to remove the white space after the `%v`

I'll add the three dots though.

Reply:
Fixed, thanks!